<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.296">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="tiffanie">
<meta name="dcterms.date" content="2023-06-15">

<title>tiffanie lab blog - Functional annotation of GWAS loci using transcriptome data</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
</style>


<script src="../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../">
<script src="../../site_libs/quarto-html/quarto.js"></script>
<script src="../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>

  <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<link rel="stylesheet" href="../../styles.css">
</head>

<body class="nav-fixed fullcontent">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="../../index.html">
    <span class="navbar-title">tiffanie lab blog</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item">
    <a class="nav-link" href="../../w1.html" rel="" target="">
 <span class="menu-text">w1</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../w2.html" rel="" target="">
 <span class="menu-text">w2</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../w3.html" rel="" target="">
 <span class="menu-text">w3</span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://github.com/tiffanieh" rel="" target=""><i class="bi bi-github" role="img">
</i> 
 <span class="menu-text"></span></a>
  </li>  
</ul>
            <div class="quarto-navbar-tools">
</div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<header id="title-block-header" class="quarto-title-block default page-columns page-full">
  <div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-body">
      <h1 class="title">Functional annotation of GWAS loci using transcriptome data</h1>
                                <div class="quarto-categories">
                <div class="quarto-category">presentations</div>
                <div class="quarto-category">w2</div>
              </div>
                  </div>
  </div>
    
  
  <div class="quarto-title-meta">

      <div>
      <div class="quarto-title-meta-heading">Author</div>
      <div class="quarto-title-meta-contents">
               <p>tiffanie </p>
            </div>
    </div>
      
      <div>
      <div class="quarto-title-meta-heading">Published</div>
      <div class="quarto-title-meta-contents">
        <p class="date">June 15, 2023</p>
      </div>
    </div>
    
      
    </div>
    
  
  </header><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    
<!-- main -->
<main class="content quarto-banner-title-block" id="quarto-document-content">




<p>Lecture notes from quant genomics training (day 2)</p>
<p>PredixCan (TWAS)</p>
<p>GTEx consortium</p>
<ul>
<li>collect 1000 organ donors and tissue samples from different sites of the body; all RNA-sequenced (have genotype data)</li>
</ul>
<p>eQTL</p>
<ul>
<li>→ that data used to calculate eQTL (expression quantitative trait loci) — SNPs associated/correlated w/ expression levels of a gene</li>
<li><span class="math inline">\(T = \gamma \cdot X+\epsilon\)</span>
<ul>
<li>X = genotype; gamma = effect size being estimated</li>
</ul></li>
</ul>
<p>we have genotype data to train predictors</p>
<ul>
<li><p>different layers of genotype matrices make up transcriptome</p></li>
<li><p>fit linear model to predict weight that gives us best representation of genetic component of expression</p></li>
<li><p>take genotype matrix → transform/impute transcriptome</p>
<p><img src="images/haky_lecture_fig1.png" class="quarto-discovered-preview-image img-fluid"></p></li>
<li><p>take genotype data → predict transcriptome</p></li>
</ul>
<p>How well are we predicting?</p>
<ul>
<li><p>compare predicted and observed expression</p></li>
<li><p>pretty high correlation (promising)</p>
<p><img src="images/haky_lecture_fig2.png" class="img-fluid"></p></li>
<li><p>black curve: measure of heritability of the expression level of that gene</p></li>
</ul>
<p>Advantages of gene level association</p>
<ul>
<li>reduced multiple testing burden (from a million to around 20k)</li>
<li>function of genes are much better annotated than SNPs</li>
<li>validation in other model systems is possible (like in mice or zebrafish)</li>
<li>reverse causality issues is less of an issue (germline DNA doesn’t change w/ disease status)</li>
<li>provides direction of effects; i.e.&nbsp;whether up or down regulation of a gene increases risk of a disease</li>
<li>candidate causal gene is a good target for drug development</li>
</ul>
<p>S-PrediXcan (summary-PrediXcan)</p>
<p>regular PrediXcan: take genotype matrix (hard to get access to) → predict transcriptome → run association for every gene → get gene-level results</p>
<ul>
<li>hard to do since it’s hard to get the data in genotype matrix</li>
</ul>
<p><img src="images/haky_lecture_fig3.png" class="img-fluid"></p>
<p>solution: S-prediXcan — take SNP-level results and get gene-level results</p>
<ul>
<li>can derive formula that gives you the gene level results from SNP-level results</li>
</ul>
<p>Limitations of TWAS methods</p>
<p><img src="images/haky_lecture_fig4.png" class="img-fluid"></p>
<ol type="1">
<li>LD contamination
<ol type="1">
<li>in TWAS, taking a bunch of eQTLs and predict mRNA to some degree (correlating that to disease level)</li>
<li>eQTL is nearby a SNP that is causing the disease — so not actually from eQTL; just from proximity (artifact of LD)</li>
<li>solutions:
<ol type="1">
<li>can use fine-mapped predictors (improve quality of list of genes associated with disease)</li>
<li>post-filtering w/ colocalization methods</li>
</ol></li>
</ol></li>
<li>Co-regulation: cannot figure out which gene is the closer one
<ol type="1">
<li>might find 2 different genes that are associated with a disease but one of them is not actually (just a confounder)</li>
<li>more difficult to tackle</li>
</ol></li>
</ol>
<p>Take home message</p>
<ul>
<li>gene-level association methods (TWAS methods) have many advantages but can lead to false positives</li>
<li>significant genes are excellent hypothesis generators that need to be confirmed with independent lines of evidence (smaller list of genes → less hypothesis tests to run)</li>
</ul>
<p>Colocalization</p>
<ul>
<li><p>given a GWAS study and eQTL study, can try to find which one is the causal gene</p></li>
<li><p>attempts to determine whether causal variants altering gene expression levels are same as causal variants that alter disease</p></li>
<li><p>key: fine-mapping to find causal variants</p>
<ul>
<li>given a statistical method, tries to find causal variants (i.e.&nbsp;what’s the probability that this SNP is the causal variant)</li>
</ul></li>
<li><p>ex. (note: normally we don’t know ground truth) true effects</p>
<img src="images/haky_lecture_fig5.png" class="img-fluid" alt="Screenshot 2023-06-15 at 9.26.43 AM.png">
<ul>
<li><p>simulate phenotype based on this model and run a GWAS</p></li>
<li><p>below: GWAS p-values</p>
<p><img src="images/haky_lecture_fig6.png" class="img-fluid"></p></li>
<li><p>higher up on the plot —&gt; more significant correlation</p>
<ul>
<li>not all points at the bottom</li>
</ul></li>
</ul></li>
<li><p>fine-mapping</p>
<p><img src="images/haky_lecture_fig7.png" class="img-fluid"></p>
<ul>
<li>PIP (posterior inclusion probability) - probability that this given variant is the causal variant using SuSi</li>
</ul></li>
</ul>
<p>Colocalization: are causal variants = ?</p>
<ul>
<li>tries to tell you whether causal SNP for one thing is same as causal signal in something else</li>
<li>ex. what’s mediating trait in FTO muscle and BMI → if we find something, can manipulate levels in one thing to see if it does anything to the other thing</li>
<li>even if location is the same the SNP could be different</li>
<li>not colocalized if signal seem to be in the same region but if you do fine mapping that the causal variant for one thing is distinct from the causal variant for the other thing</li>
</ul>
<p>Limits of colocalization</p>
<ul>
<li>fine-mapping can be sensitive to LD reference
<ul>
<li><p>correlation matrix that we use between SNPs</p>
<p><img src="images/haky_lecture_fig8.png" class="img-fluid"></p></li>
</ul></li>
</ul>
<p>Other TWAS methods</p>
<ul>
<li>TWAS/FUSION</li>
<li>UTMOST</li>
<li>cTWAS
<ul>
<li>uses both fine mapping and association to calculate probability of gene being causal accounting for LD</li>
</ul></li>
</ul>
<p>Colocalization Methods</p>
<ul>
<li>COLOC</li>
<li>ENLOC</li>
<li>fastENLOC</li>
<li>eCAVIAR</li>
</ul>
<p>Mendelian randomization</p>
<p>Randomized trial vs Mendelian randomization</p>
<p><img src="images/haky_lecture_fig9.png" class="img-fluid"></p>
<ul>
<li>Mendelian randomization: uses genetic variant as instruments to try to find causal exposures</li>
<li>randomized trial
<ul>
<li>ex. in biomedical studies if you want to know the effect of a drug on the severity of a disease, need to run randomized trial (control+treatment) in order to determine causality</li>
</ul></li>
<li>Mendelian
<ul>
<li>randomization occurs by genetic variant (risk allele present or absent) aka nature’s randomized trial</li>
</ul></li>
</ul>
<p><img src="images/haky_lecture_fig10.png" class="img-fluid"></p>
<ul>
<li>level of evidence for causation is much higher in Mendelian randomization</li>
</ul>
<p>Mendelian randomization question</p>
<ul>
<li><p>goal: test association between a modifiable exposure+disease (ex. smoking, HDL cholesterol levels, etc.)</p></li>
<li><p>problem: if you just look at exposure and disease there could be confounders</p></li>
<li><p>solution: IV (SNPs)</p>
<ul>
<li>3 IV assumptions:
<ul>
<li>independence assumption between SNPs and confounders</li>
<li>relevance assumption between SNPs and exposure (reliable association)</li>
<li>exclusion restriction assumption between SNPs and disease (no direct effect)</li>
</ul></li>
</ul></li>
<li><p>TWAS can be thought of as Mendelian randomization</p>
<ul>
<li>“exposure” — baseline expression of gene</li>
<li>want to see if up or down regulation of gene is causing a disease</li>
<li>if we use SNPs to predict expression level of gene as instrument, can investigate any associations</li>
</ul></li>
<li><p>SMR (summary data-based Mendelian randomization)</p>
<p><img src="images/haky_lecture_fig11.png" class="img-fluid"></p></li>
<li><p>same as previous way but with eQTL as IV</p>
<p><img src="images/haky_lecture_fig12.png" class="img-fluid"></p></li>
<li><p>multiple eQTL and multiple genes</p></li>
</ul>
<p>New methods</p>
<ul>
<li><p>MetaboXcan: predict metabolite levels</p>
<p><img src="images/haky_lecture_fig13.png" class="img-fluid"></p>
<p><img src="images/haky_lecture_fig14-03.png" class="img-fluid"></p>
<ul>
<li>added link between mRNA levels and metabolites (don’t just want list of genes and list of metabolites)</li>
</ul></li>
<li><p>Deep learning-based predictors</p>
<p><img src="images/haky_lecture_fig15.png" class="img-fluid"></p>
<p><img src="images/haky_lecture_fig16.png" class="img-fluid"></p>
<p><img src="images/haky_lecture_fig17.png" class="img-fluid"></p>
<ul>
<li>TFXcan</li>
<li>single cell PrediXcan</li>
<li>EpigenomeXcan</li>
</ul></li>
</ul>



</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>