<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.296">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="tiffanie">
<meta name="dcterms.date" content="2023-06-22">

<title>tiffanie lab blog - Enformer paper summary</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
</style>


<script src="../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../">
<script src="../../site_libs/quarto-html/quarto.js"></script>
<script src="../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>

  <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<link rel="stylesheet" href="../../styles.css">
</head>

<body class="nav-fixed fullcontent">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="../../index.html">
    <span class="navbar-title">tiffanie lab blog</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item">
    <a class="nav-link" href="../../w1.html" rel="" target="">
 <span class="menu-text">w1</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../w2.html" rel="" target="">
 <span class="menu-text">w2</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../w3.html" rel="" target="">
 <span class="menu-text">w3</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../w4.html" rel="" target="">
 <span class="menu-text">w4</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../w5.html" rel="" target="">
 <span class="menu-text">w5</span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://github.com/tiffanieh" rel="" target=""><i class="bi bi-github" role="img">
</i> 
 <span class="menu-text"></span></a>
  </li>  
</ul>
            <div class="quarto-navbar-tools">
</div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<header id="title-block-header" class="quarto-title-block default page-columns page-full">
  <div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-body">
      <h1 class="title">Enformer paper summary</h1>
                                <div class="quarto-categories">
                <div class="quarto-category">readings</div>
                <div class="quarto-category">w3</div>
              </div>
                  </div>
  </div>
    
  
  <div class="quarto-title-meta">

      <div>
      <div class="quarto-title-meta-heading">Author</div>
      <div class="quarto-title-meta-contents">
               <p>tiffanie </p>
            </div>
    </div>
      
      <div>
      <div class="quarto-title-meta-heading">Published</div>
      <div class="quarto-title-meta-contents">
        <p class="date">June 22, 2023</p>
      </div>
    </div>
    
      
    </div>
    
  
  </header><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    
<!-- main -->
<main class="content quarto-banner-title-block" id="quarto-document-content">




<p><a href="https://www.nature.com/articles/s41592-021-01252-x">paper link</a></p>
<p><strong>objective:</strong> improve gene expression prediction accuracy from DNA sequences using a type of DL architecture (Enformer) that can include info from long-range interactions in genome → higher accuracy for variant effect predictions on gene expression, learned to predict enhancer-promoter interactions directly from DNA sequence → future steps: more effective fine-mapping of human disease associations</p>
<p><strong>Main</strong></p>
<p>Population-based association studies+models predicting gene expression+chromatin states from DNA sequences:</p>
<ul>
<li>the former limited to common variants; hard to disentangle causality from assoc due to LD (linkage disequilibrium)</li>
<li>also hard to test all variants of interest in different relevant contexts in experimental validation</li>
<li>sequence-based computational models have limited accuracy</li>
</ul>
<p>Currently: deep CNNs to predict gene expr from DNA seqs for human+mouse genomes</p>
<ul>
<li>drawback: can only consider seq elements &lt; 20 kb away from TSS (transcription start site) b/c locality of convolutions limits info flow in net between far apart elements
<ul>
<li>many regulatory elements can infleunce gene expr from far away so need to increase info flow between distal elements</li>
</ul></li>
</ul>
<p><strong>Results</strong></p>
<p>Solution: <strong>Enformer</strong> — a new model architecture to predict gene expr+chromatin states in humans+mice from DNA seqs</p>
<ul>
<li>transformers: a class of DL models for NLP; recently applied to model short DNA seqs
<ul>
<li>consist of attention layers that transform each position in input sequence by computing weighted sum across representations of all other positions in sequence</li>
<li>attention weight between any 2 positions depends on embeddings of current representation vectors+distance between</li>
<li>model can refine prediction at a TSS by getting info from relevant regions (e.g.&nbsp;relevant enhancers)</li>
<li>better info flow between distal elements since each position directly attends to all other positions in sequence</li>
</ul></li>
<li>vs convolutional layers — need many many layers to get to far away elements due to their local receptive field</li>
<li><strong>transformer layers → increase receptive field → reach distal regulatory elements up to 100 kb away</strong>
<ul>
<li>expands # relevant enhancers seen by model from 47% to 84% (&lt;100 kb)</li>
</ul></li>
<li>Enformer better at predicting RNA expr at TsS of human protein-coding genes, consistent across all 4 types of genome wide tracks (CAGE — transcriptional activity, histone modifications, TF binding, DNA accessibility)
<ul>
<li>esp CAGE — prob b/c tissue-specific gene expr depends a lot on distal elements</li>
</ul></li>
<li>greater predictive accuracy than ExPecto (model trained to predict gene expr levels measured by RNA seq) for across-genes+across-tissues</li>
</ul>
<p>Demonstrating benefit of attention layers</p>
<ul>
<li>(compared to dilated convolutions in previous model)</li>
<li>replace attention layers w/ dilated convolutions+tuned learning rate for optimal performance → attention layers outperformed dilated convolutions across all model sizes, # layers, # training data pts</li>
<li>observed performance drop when restricting receptive field of Enformer to that of the previous model by replacing global attention layers w/ local</li>
<li>increase # params → improve model performance (consistent w/ recent advances in NLP)</li>
<li>use <strong>custom relative positional basis fns</strong> in transformer layers (distinguish between proximal and distal regulatory elements more easily)
<ul>
<li>noticeable performance improvement compared to typically used relative basis fns+absolute positional encodings in NLP lit</li>
</ul></li>
</ul>
<p>Enformer attends to cell-type-specific enhancers</p>
<ul>
<li>what seq elements Enformer uses to make predictions? → compute 2 diff gene expr contribution scores:
<ol type="1">
<li>input gradients (gradient <span class="math inline">\(\times\)</span> input)</li>
<li>attention weights</li>
</ol>
for several genes w/ CRISPRi-validated enhancers</li>
</ul>
<p>contribution scores — highlight most predictive input seq for particular gene’s expression</p>
<ul>
<li>input gradients are tissue- or cell-type-specific (b/c computed wrt particular output CAGE sample)</li>
<li>vs attention weights — internal to model+shared among all tissue+cell-type predictions</li>
<li>observed that contribution scores of several genes correlated w/ histone H3 acetylated at K27 (H3K27ac) and highlighted not just local promoter regions but also distal enhancers &gt;20 kb away
<ul>
<li>suggests that Enformer is looking at biologically relevant regions when making predictions; <strong>gene expr contribution scores could be used to prioritize</strong> relevant enhancers</li>
</ul></li>
<li>unsolved problem: how to link candidate enhancers id’ed vai biochemical annotations to target genes
<ul>
<li>issue: computational models have historically low accuracy b/c of noisy labels+class imbalance</li>
</ul></li>
<li>evaluate ability of contribution scores to pinpoint relevant enhancers for a specific gene: compare several contribution scores across all tested enhancer-gene pairs in 2 large-scale CRISPRi studies performed on K562 cell line</li>
<li>CRISPRi used to suppress activity of &gt; 10,000 candidate enhancers+measure effect on gene expr</li>
<li>contribution scores prioritize validated en-gene pairs w/ higher accuracy despite only using DNA seq as input — never trained to explicity locate enhancers
<ul>
<li>can be used for arbitrary seq variations lacking experimental data</li>
</ul></li>
<li>model uses diff enhancer seqs in diff cell types as expected → can prioritize candidate enhancers in cell types used for model training</li>
</ul>
<p>has model learned about insulator elements (separate 2 topologically associating domains, i.e.&nbsp;<strong>TADs</strong>, minimize enhancer-promoter crosstalk between the two)</p>
<ul>
<li>used attention matrices (more efficient than input gradients to compute b/c of many output targets) of seqs centered at TAD boundaries+compared to attention from seqs w/ no particular alignment</li>
<li>Enformer paid more attention to TAD boundaries than random positions; less attention to regions on opposite side of boundary — consistent w/ reduced inter-TAD interactions in biology</li>
<li>suggest that model has not only learned about role of tissue-specific enhancers+promoters, but also about insulator elements (and their role in inhibiting info flow between genomic compartments)</li>
</ul>
<p>Enformer improves variant effect prediction on eQTL data</p>
<ul>
<li>central goal: predict influence of genetic variants on cell-type-specific gene expr
<ul>
<li>→ inform fine-mapping of many noncoding assoc w/ phenotypes of interest from GWAS</li>
</ul></li>
<li>computational models that predict regulatory activity from DNA seqs can process distinct alleles+compare preds to score genetic variants</li>
<li>successful model can produce eQTL results w/o having to measure lots of individual gene expr profiles</li>
<li>challenge: LD in profiled pop → transfers causal eQTL’s effect to nearby co-occurring variants’ measurements</li>
<li>SLDP (<strong>signed linkage disequilibrium profile</strong>) regression: measures genome-wide statistical concordance between signed variant annotations (e.g.&nbsp;model preds) and GWAS summary stats (e.g.&nbsp;GTEx eQTLs) while accounting for LD</li>
<li>Enformer preds for noncoding-variant activity seem to improve for samples w/ similar cell-type composition mostly</li>
</ul>
<p>how to assess use of preds for identifying causal variants → classification task per tissue → discriminate likely causal variants from likely spurious eQTLs</p>
<ul>
<li>each variant represented by its pred diff vector (reference-alternative allele, summed across seq) for all 5313 human datasets</li>
<li>trained random forest classifiers</li>
<li>preds enabled more accurate classifier for most GTEx tissues</li>
</ul>
<p>Enformer improves MPRA mutation effect prediction</p>
<ul>
<li>eval Enformer’s performance on second, independent variant effect pred tas w/ dataset where MPRAs (massively parallel reporter assays) directly measured fnal effect of genetic variatns through sutration mutagenesis of several enhancers+promoters in variety of cell types</li>
<li>for each variant, eval effect as pred diff between ref+alt allele → retrieve 5313 features</li>
<li>compare 2 approaches:
<ol type="1">
<li>train lasso reg model on provided training set for each gene</li>
<li>preselect subset of features corresponding to cell-type-matched/agnostic preds of changes in CAGE+DNase → generate summary stat of features</li>
</ol></li>
</ul>
<p><strong>Lasso regression</strong> w/ Enformer preds as features had best average correlation across all loci</p>
<p><strong>Future steps</strong></p>
<ul>
<li>systematically apply Enformer to fine-map existing GWAS studies</li>
<li>prioritize rare or de novo variants for rare disorders</li>
<li>impute regulatory activity across species to study <strong>cis</strong>-regulatory evolution</li>
</ul>
<p><strong>Methods</strong></p>
<p>Model architecture</p>
<p>diagram of typical Transformer architecture:</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="https://machinelearningmastery.com/wp-content/uploads/2021/08/attention_research_1-727x1024.png" class="quarto-discovered-preview-image img-fluid figure-img"></p>
<p></p><figcaption class="figure-caption">https://machinelearningmastery.com/wp-content/uploads/2021/08/attention_research_1-727x1024.png</figcaption><p></p>
</figure>
</div>
<p>Enformer architecture has 3 parts</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="https://www.biorxiv.org/content/biorxiv/early/2021/04/08/2021.04.07.438649/F5.medium.gif" class="img-fluid figure-img"></p>
<p></p><figcaption class="figure-caption">https://www.biorxiv.org/content/biorxiv/early/2021/04/08/2021.04.07.438649/F5.medium.gif</figcaption><p></p>
</figure>
</div>
<ol type="1">
<li><strong>7 convolutional blacks w/ pooling</strong></li>
<li><strong>11 transformer blocks</strong></li>
<li>a <strong>cropping layer</strong> followed by <strong>final pointwise convolutions branching into 2 organism-specific network heads</strong></li>
</ol>
<p><strong>input:</strong> one-hot-encoded DNA seq</p>
<ul>
<li>A = <span class="math inline">\(\left[1,0,0,0\right]\)</span></li>
<li>C = <span class="math inline">\(\left[0,1,0,0\right]\)</span></li>
<li>G = <span class="math inline">\(\left[0,0,1,0\right]\)</span></li>
<li>T = <span class="math inline">\(\left[0,0,0,1\right]\)</span></li>
<li>N = <span class="math inline">\(\left[0,0,0,0\right]\)</span></li>
</ul>
<p>of length 196,608 bp → predicts 5313 genomic tracks for human genome, 1643 tracks for mouse genome (each of length 896 corresponding to 114,688 bp aggregated into 128-bp bins)</p>
<ul>
<li>convolutional blocks w/ pooling first reduce the spatial dimension from 196,608 bp to 1526 so each seq position vector represents 128 bp</li>
<li>transformer blocks capture long-range interactions across seq</li>
<li>cropping layer trims 320 positions on each</li>
<li>to avoid computing loss on far ends b/c these regions are disadvantage (observe regulatory elements only on one side, toward seq center)</li>
<li>2 output heads predict organism-specific tracks</li>
</ul>
<p>changes: use <strong>transformer blocks</strong> instead of dilated convolutions, <strong>attention pooling</strong> instead of max pooling, twice as many channels, 1.5x longer input seq</p>
<p><strong>attention pooling:</strong> summarizes contiguous chunk of input sequence <span class="math inline">\(\mathbf{x}_{k:k+L_p}^{full}=\mathbf{x} \in R^{L_p\times C}\)</span> across <span class="math inline">\(L_p\)</span> positions for each of the <span class="math inline">\(C\)</span> chanels</p>
<ul>
<li>output value <span class="math inline">\(\mathbf{h} \in \mathbf{\mathit{R}}^C\)</span>:</li>
</ul>
<p><span class="math display">\[
h_j = \frac{\sum_i exp(x_i \cdot w_j)x_{ij}}{\sum_i exp(x_i\cdot w_j)}
\]</span></p>
<p>where <strong>i</strong> indexes seq position in pooling window, which is weighted by exponentiated dot product <span class="math inline">\(\mathbf{x}_i\cdot \mathbf{w}_j\)</span> and <span class="math inline">\(\mathbf{\mathit{w}} \in R^{C\times K}\)</span> is a matrix of learned weights</p>
<ul>
<li>apply attention pooling to contiguous chunks of original input seq using window size <span class="math inline">\(L_p = 2\)</span> and stride of 2</li>
</ul>
<p>initialize <span class="math inline">\(\mathbf{w}\)</span> to <span class="math inline">\(2\times \mathbf{1}\)</span></p>
<ul>
<li><span class="math inline">\(\mathbf{1}\)</span> = identity matrix</li>
<li>prioritizes larger value → similar operation to max pooling</li>
<li>slightly better performance than random initialization or initialization w/ zeros (average pooling)</li>
</ul>
<p>MHA (<strong>multi-head attention</strong>) layers used to capture long-range interactions across sequence</p>
<ul>
<li>each head has separate weights for queries, keys, values that transform input seq
<ul>
<li>separate set of weights <span class="math inline">\(\mathbf{w}^q \in \mathbf{R}^{C\times K}\)</span>, <span class="math inline">\(\mathbf{w}^k \in \mathbf{R}^{C\times K}\)</span>, <span class="math inline">\(\mathbf{w}^v \in \mathbf{R}^{C\times K}\)</span></li>
<li>→ transform input sequence <span class="math inline">\(\mathbf{x}\in \mathbf{R}^{L\times C}\)</span> into:
<ul>
<li>queries <span class="math inline">\(\mathbf{q}_i=\mathbf{x}_i\mathbf{w}^q\)</span></li>
<li>keys <span class="math inline">\(\mathbf{k}_j=\mathbf{x}_j\mathbf{w}^k\)</span></li>
<li>values <span class="math inline">\(\mathbf{v}_j=\mathbf{x}_j\mathbf{w}^v\)</span></li>
</ul></li>
</ul></li>
<li>queries = current info at each position</li>
<li>keys = info each position looks for to attend to</li>
<li><strong>dot product between queries+keys</strong> plus relative positional encodings <span class="math inline">\(\mathbf{R}_{ij}\)</span> → attention matrix
<ul>
<li>computed as <span class="math inline">\(\mathbf{a}_{ij} = softmax(\mathbf{q}_i\mathbf{k}_j^T/\sqrt{K}+\mathbf{R}_{ij})\)</span></li>
<li>entry <span class="math inline">\(a_{ij}\)</span> = amount of weight that query at position <strong>i</strong> puts on key at position <strong>j</strong></li>
</ul></li>
<li>values = info propagated to positions attending to a specific position</li>
<li>each single attention head computes its output as weighted sum across all input positions: <span class="math inline">\(\mathbf{av}\)</span>
<ul>
<li>lets query position use info across whole seq</li>
<li>multiple heads compute w/ independent parameters</li>
</ul></li>
<li>outputs from each head concatenated to form final layer output followed by linear layer to combine</li>
</ul>
<p>Enhancer prioritization:</p>
<ol type="1">
<li><strong>Gradient <span class="math inline">\(\times\)</span> input:</strong> compute abs value of gradient of CAGE targets at TSS w. regard to input reference seq nucleotide
<ol type="1">
<li>equivalent to computing gradient <span class="math inline">\(\times\)</span> input attirbutions (since one-hot-encoding)</li>
<li>CAGE at TSS = summing abs grad values from 3 adjacent bins (including bin overlapping TSS, one flanking bin on each side)</li>
<li>enhancer-gene scare obtained by summing abs grad<span class="math inline">\(\times\)</span>input scores in 2-kb window centered at enhancer</li>
</ol></li>
<li><strong>attention:</strong> averaged transformer attention matrices across all heads+layers → extract row corresponding to query index positioned at TSS
<ol type="1">
<li>keys correspond to diff spatial positions</li>
<li>attention values = how much model attended to these positions when making preds for TSS</li>
<li>en-gene score — sum attention scores in 2-kb window centered at enhancer</li>
</ol></li>
<li><strong>ISM:</strong> <em>in silico</em> mutagenesis en-gene score
<ol type="1">
<li>computed by comparing K562 CAGE preds at TSS from ref seq with preds from modif seq where 2-kb enhancer seq was replaced by a random sequence: <span class="math inline">\(|f(modified)-f(reference)|\)</span></li>
</ol></li>
</ol>
<p>other details for methodology on model training+eval can be found in paper*</p>



</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>